compiler/src/iree/compiler/GlobalOptimization/test/set_encoding.mlir

```diff
@@ -931,118 +931,6 @@ func.func @batch_matmul_truncf_f16f16f32(%arg0 : tensor<64x100x250xf32>, %arg1 :
 
 // -----
 
-func.func @batch_matmul_casted_ui8i8i32(%arg0 : tensor<64x100x250xi8>, %arg1 : tensor<64x250x500xi8>,
```

Comment:
Keep the existing test cases! Do add your new testcase alongside. We want to make sure that we keep data-tiling named ops like this the way we used to, while gaining the ability to also data-tile generics.

Reply:
This test case was for the separate element-wise cast producers that we were using to handle signed vs. unsigned semantics of matmuls. We no longer need to do this now that we can directly set encodings on linalg.generic ops. My impression from when we added this workaround a while ago was that we wanted to eventually remove it when possible because it is a bit messy to be tying 2 ops together through the whole compiler flow. I feel that this particular encoding is very specific to that one workaround (as it is specific to element-wise castOpInterface ops), and we probably don't want to keep it around longer than we need.

With https://github.com/openxla/iree/pull/16164, we will not see any cases like this anymore, so maybe it would be better to remove that pass in this PR. I don't know if any cases previously existed (before we started creating them) where there is an element-wise cast preceding a matmul, but I think we would want to handle cases like that by fusing the ops into the matmul (rather than setting a separate encoding on them). What do you think? If you think it is better to just keep the functionality then I don't have any issues with adding it back.

Reply:
Oh.. OK. No strong opinion either way, as long as you're confident that we're not regressing anything here. It's true that we did add this for one specific llama2 benchmark that is what you're still looking at, so if you know this doesn't regress, that's probably good enough.